Make the Labeling guide easier to follow

ptgott opened the pull request:
See #11841

The current guide is a conceptual introduction to Teleport's labeling
functionality. Since we cover this information in our reference guides,
I wanted to make it possible for readers to follow this guide as a
tutorial and get started labeling resources quickly.

- Introduce labels at the beginning of the guide to set the stage
  conceptually before the steps begin.
- Clarify in the Prerequisites section that the instructions in the
  guide apply to any Teleport resource, not just Nodes
- Require a host for running a Teleport Node in the Prerequisites and
  include instructions for starting the Node. Previously, we included
  sample commands/config for applications as well. Requiring a Node
  and including setup instructions for that Node will ensure that all
  instructions in the guide can be followed easily.
- Add scoped instructions via Tabs and ScopedBlocks.
- Reorganize sections into a series of steps.
- Add commands to verify that labels have been added.

ptgott committed:
Make the Labeling guide easier to follow

See #11841

The current guide is a conceptual introduction to Teleport's labeling
functionality. Since we cover this information in our reference guides,
I wanted to make it possible for readers to follow this guide as a
tutorial and get started labeling resources quickly.

- Introduce labels at the beginning of the guide to set the stage
  conceptually before the steps begin.
- Clarify in the Prerequisites section that the instructions in the
  guide apply to any Teleport resource, not just Nodes
- Require a host for running a Teleport Node in the Prerequisites and
  include instructions for starting the Node. Previously, we included
  sample commands/config for applications as well. Requiring a Node
  and including setup instructions for that Node will ensure that all
  instructions in the guide can be followed easily.
- Add scoped instructions via Tabs and ScopedBlocks.
- Reorganize sections into a series of steps.
- Add commands to verify that labels have been added.

r0mant commented on docs/pages/setup/admin/labels.mdx:
```diff
@@ -1,66 +1,274 @@
 
-This guide explains how to label Teleport Nodes.
+Teleport allows you to add arbitrary key-value pairs to your applications,
+Nodes, databases, and other resources in your cluster. You can use these to
```
Nit: Looks weird that only "Nodes" is capitalized.

r0mant commented on docs/pages/setup/admin/labels.mdx:
```diff
@@ -1,66 +1,274 @@
-This guide explains how to label Teleport Nodes.
+Teleport allows you to add arbitrary key-value pairs to your applications,
+Nodes, databases, and other resources in your cluster. You can use these to
+filter resource when making queries via `tctl` and `tsh` as well as to limit the
```
```suggestion
filter resources when making queries via `tctl` and `tsh` as well as to limit the
```

ptgott committed:
Respond to PR feedback

zmb3 commented on docs/pages/setup/admin/labels.mdx:
```diff
@@ -1,66 +1,274 @@
-  configurable frequency and the output of the command becomes the label value.
-  Examples include reporting load averages, the presence of a process, and the time after
-  the last reboot.
+You can label any resource managed by Teleport using the instructions in this
+guide. See the following guides for how to add resources to a Teleport cluster:
```
Labeling works differently for Windows desktops (there is no dynamic labels because there is no Teleport software running on the desktop). `docs/pages/desktop-access/rbac.mdx` has a labeling section that would be good to link to in the further reading section.

Databases are also a bit different when discovered automatically.

zmb3 commented on docs/pages/setup/admin/labels.mdx:
```diff
@@ -1,66 +1,274 @@
+$ tsh ls
+Node Name         Address    Labels          
+----------------- ---------- --------------- 
+ip-192-0-2-0 ⟵ Tunnel   kernel=5.13.0-1021-aws,uptime=up 40 minutes 
```
Looks like the whitespace is off a bit here.
```suggestion
ip-192-0-2-0.     ⟵ Tunnel   kernel=5.13.0-1021-aws,uptime=up 40 minutes 
```

zmb3 commented on docs/pages/setup/admin/labels.mdx:
````diff
@@ -1,66 +1,274 @@
+Start Teleport:
+
+```code
+$ sudo teleport start \
+  --auth-server=${ADDR?}\
+  --roles=node \
+  --token=${TOKEN?} \
+  --labels uptime=[1m:"uptime -p"],kernel=[1h:"uname -r"]
````
We are trying to get away from recommending long `teleport start` commands with lots of complicated flags in favor of simpler `teleport start -c teleport.yaml` start commands - thus pushing the configuration into a file.

Given this, and the fact that this syntax for specifying labels on the command line is pretty complicated, I would prefer we just recommend the config file approach here too.

ptgott committed:
Improve the Labeling guide for self-hosted users

See: #11841

- Edit tctl.mdx: Users of self-hosted Teleport can run tctl commands
  from their local machines after logging in to a cluster, just like
  they can with Teleport Cloud. I have edited tctl.mdx to instruct self-
  hosted users to use tctl remotely. This way, users won't need to
  access their Auth Service hosts in order to follow the guides that
  use tctl.mdx.

- Add steps for fixing unexpected "tsh ls" output

Also applies PR feedback.

- Add provisions for labeling specific resources as a table near the
  top of the page.
- Remove the option to add labels using CLI flags.
- Fix command output spacing

Also misc. clarity tweaks.

ptgott commented:
I am going to follow this guide with  a self-hosted cluster, and will make some more changes while incorporating the current PR feedback. I will turn this into a draft PR until the changes are ready for review.

ptgott commented:
@r0mant I've changed this guide pretty substantially since your last review, so I'd appreciate another review if you have time. Thanks!

strideynet reviewed (commented):
Looks mostly good, just one question and I think one place where we can use more consistency language.

ptgott commented on docs/pages/setup/admin/labels.mdx:
```diff
@@ -1,66 +1,268 @@
+- **Dynamic labels**, also known as **label commands**, allow you to generate
+  labels at runtime. Teleport will execute an external command on a Node at a
+  configurable frequency and the output of the command becomes the label value.
+  Examples include reporting load averages, the presence of a process, and the time after
```
This text was there before I changed it, so I worked with the assumption that it was there for a reason. That's a good point, though—I've suggested something more realistic.

strideynet replied:
That's a really nice relevant example you've added !

ptgott committed:
Respond to PR feedback
